mass_mailing_custom_unsubscribe/controllers/main.py

```diff
@@ -53,23 +55,38 @@ def mailing(self, mailing_id, email=None, res_id=None, token="", **post):
             # Unsubscribe, saving reason and details by context
             details = post.get("details", False)
             self._add_extra_context(mailing_id, res_id, reason_id, details)
-            # You could get a DetailsRequiredError here, but only if HTML5
-            # validation fails, which should not happen in modern browsers
-            result = super().mailing(
-                mailing_id, email, res_id, token=token, **post)
-            result.qcontext.update({"reasons": reasons})
-            # update list_ids taking into account not_cross_unsubscriptable
-            # field
             mailing_obj = request.env['mail.mass_mailing']
-            mailing = mailing_obj.sudo().browse(mailing_id)
-            if mailing.mailing_model_real == 'mail.mass_mailing.contact':
-                result.qcontext.update({
-                    "list_ids": result.qcontext["list_ids"].filtered(
-                        lambda mailing_list:
-                            not mailing_list.not_cross_unsubscriptable or
-                            mailing_list in mailing.contact_list_ids
-                    )
+            mass_mailing = mailing_obj.sudo().browse(mailing_id)
+            model = mass_mailing.mailing_model_real
+            if "opt_out" in request.env[model]._fields:
+                mass_mailing.update_opt_out_other(email, [res_id], True)
+                result = request.render("mass_mailing.page_unsubscribed", {
+                    "email": email,
+                    "mailing_id": mailing_id,
+                    "res_id": res_id,
+                    "show_blacklist_button": request.env[
+                        "ir.config_parameter"
+                    ].sudo().get_param(
+                        "mass_mailing.show_blacklist_buttons"
+                    ),
```

Comment:
Don't you need to cast this into bool?

```suggestion
                    "show_blacklist_button": odoo.tools.str2bool(request.env[
                        "ir.config_parameter"
                    ].sudo().get_param(
                        "mass_mailing.show_blacklist_buttons"
                    )),
```

Reply:
This code was also copied from upstream, so the problem will also be there if so. No sense to have here extra care if in upstream this will fail.